R/ARIMATimeSeries.R

```diff
@@ -0,0 +1,642 @@
+    plot$dependOn(dependencies)
+    plot$position <- position
+
+    jaspResults[["timeSeriesPlot"]] <- plot
+
+    if (!ready) {
+      return()
+    }
+
+    .tsFillTimeSeriesPlot(plot, dataset, options, type = options$timeSeriesPlotType, distribution = options$timeSeriesPlotDistribution)
+  }
+}
+
+.tsArimaResults <- function(jaspResults, dataset, options, ready, dependencies) {
+  if (!is.null(jaspResults[["arimaResult"]])) {
+    return()
+  }
+
+  if (ready) {
+    if (options$seasonal) {
+      P <- options$P
+      D <- options$D
+      Q <- options$Q
+      if (options$periodSpecification == "custom") m <- options$m
+      if (options$periodSpecification == "dominant") m <- forecast::findfrequency(dataset$y)
+    } else {
+      m <- 1
+      P <- D <- Q <- 0
+    }
+    y <- ts(dataset$y, frequency = m)
+    seasonOrder <- list(order = c(P, D, Q), period = m)
+
+    xreg <- NULL
+    if (length(options[["covariates"]]) > 0) {
+      covariates <- dataset[, grepl("xreg", names(dataset))]
+      xreg <- as.matrix(covariates)
+    }
+
+    if (options$modelSpecification == "custom") {
+      fit <- try(forecast::Arima(
+        y,
+        include.constant = options$intercept,
+        order = c(options$p, options$d, options$q),
+        xreg = xreg,
+        seasonal = seasonOrder
+      ))
+    }
+
+    if (options$modelSpecification == "auto") {
+      fit <- try(forecast::auto.arima(
+        y,
+        allowdrift = options$intercept,
+        allowmean = options$intercept,
+        ic = options$modelSpecificationAutoIc,
+        xreg = xreg,
+        seasonal = options$seasonal
+      ))
+    }
+
+    if (jaspBase::isTryError(fit)) .quitAnalysis(gettext("The ARIMA model could not be fit."))
+    if (length(fit$coef) == 0) {
+      .quitAnalysis(gettext("No parameters are estimated."))
+    }
+
+    jaspResults[["arimaResult"]] <- createJaspState(fit)
+    jaspResults[["arimaResult"]]$dependOn(dependencies)
+  }
+}
+
+.tsCreateTableModel <- function(jaspResults, fit, dataset, options, ready, position, dependencies) {
+  if (!is.null(jaspResults[["modelTable"]])) {
+    return()
+  }
+
+  table <- createJaspTable(gettext(gettext("Model Summary")))
+  table$dependOn(dependencies)
+  table$position <- position
+  table$showSpecifiedColumnsOnly <- TRUE
+
+  table$addColumnInfo(name = "sigma", title = "\u03C3\u00B2", type = "number")
+  table$addColumnInfo(name = "ll", title = gettext("Log-Likelihood"), type = "number")
+  table$addColumnInfo(name = "aicc", title = gettext("AICc"), type = "number")
+  table$addColumnInfo(name = "aic", title = gettext("AIC"), type = "number")
+  table$addColumnInfo(name = "bic", title = gettext("BIC"), type = "number")
+
+  jaspResults[["modelTable"]] <- table
+
+  # Check if ready
+  if (!ready) {
+    rows <- data.frame(
+      sigma = ".",
+      ll = ".",
+      aicc = ".",
+      aic = ".",
+      bic = "."
+    )
+    row.names(rows) <- paste0("row", 1)
+    table$addRows(rows)
```

Comment:
You don't have to do this, if the jaspTable is empty it should be already formatted this way.